lib/execution_segment_test.go

```diff
@@ -379,4 +384,66 @@ func TestExecutionSegmentStringSequences(t *testing.T) {
 	}
 }
 
-// TODO: test with randomized things
+// Return a randomly distributed sequence of numSegments amount of
+// execution segments whose length totals 1.
+func genRandomExecutionSegmentSequence(numSegments int) (ExecutionSegmentSequence, error) {
```

Comment:
@MStoykov implementation in https://github.com/loadimpact/k6/blob/9fa61d917c9f3bdfbb744ff6a40db2f851522098/lib/execution_segment_test.go#L506 looks simpler, and we should probably use one :)

Should I port his implementation here?

Reply:
yeah, his seems like it can support random denominators as well, so I think you should port it as a helper function in the tests - then he can use it, and not a lambda :smiling_imp:

Reply:
Resolved in 0a2bc9c2.